velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -41,6 +41,78 @@ Expected<std::shared_ptr<DateTimeFormatter>> getDateTimeFormatter(
 }
 } // namespace detail
 
+namespace {
+enum class DateTimeFormat {
+  kMicrosecond,
+  kMillisecond,
+  kSecond,
+  kMinute,
+  kHour,
+  kDay,
+  kWeek,
+  kMonth,
+  kQuarter,
+  kYear
+};
+
+inline std::optional<DateTimeFormat> fromDateTimeFormatString(
+    const StringView& formatString,
+    bool throwIfInvalid) {
```

Comment:
We can reuse the 'fromDateTimeUnitString' method in 'velox/functions/prestosql' by moving it 'velox/functions/lib' and perhaps add a parameter like 'allowMirco' to control different behaviors.